docs/AzDO-Example.md

```diff
@@ -0,0 +1,145 @@
+- Clone this repository and navigate to {path\_to\_folder}/src/APIM\_ARMTemplate/apimtemplate
+- Restore its packages using
+
+``` 
+dotnet restore 
+```
+
+- Make sure you have signed in using Azure CLI and have switched to the subscription containing the API Management instance from which the configurations will be extracted.
+
+```
+
+az login
+
+az account set \--subscription \<subscription\_id\>
+
+```
+
+**Extractor Arguments**
+
+You have two choices when specifying your settings:
+
+1\. By using a json file with key-values where the keys matches the table below. Use the \`extractorConfig\` argument:
+
+\`extract \--extractorConfig c:/temp/extractSettings.json\`. [See more examples.](#extractorParameterFileExamples)
+
+2\. Pass the arguments on the command line. For instance \`extract \--sourceApimName my-feature-apim \--destinationApimName company-stable-apim \--resourceGroup my-feature-rg \--fileFolder c:\\temp\\apim-extract \--apiName MyFeatureV1Api\`.
+
+For this example, we will only use the option \#2
+
+So, run the application with:
+
+```
+
+dotnet run extract --sourceApimName <DEV-APIM-NAME> --destinationApimName <DESTINATION-APIM-NAME> --resourceGroup <RESOURCE-GROUP-NAME> --fileFolder c:\\temp\\apim-extract.
+
+```
+
+Where:
+
+**DEV-APIM-NAME:** API Management where you created your API
+
+**DESTINATION-APIM-NAME:** It\'s just a convention to set the DESTINATION-API-NAME in front of the generated files.
+
+**RESOURCE-GROUP-NAME:** Resource group where the DEV-APIM-NAME is hosted.
+
+***For more information on how to run the application and parameters, [go to this page](https://github.com/Azure/azure-api-management-devops-resource-kit/blob/master/src/APIM_ARMTemplate/README.md#extractor).***
+
+After executing the command above, you will see something similar to this:
+
+![A screenshot of text Description automatically
+generated](images/extractor.png)
+
+Then you see the json files extracted:
+
+![A close up of text on a white background Description automatically
+generated](images/extracted-files.png)
+
+Now, push them to your Azure DevOps Repo
+
+```
+git push
+```
+
+# Build Pipeline
+
+We will use a **Build Pipeline** to pull the extracted files from a repo and put it on Azure DevOps Artefacts Folder. 
+ 
+1 - Add these tasks to your build:
+- Get Sources
+- Copy Publish Artifacts
+
+2 - Configure the fields as show bellow:
+
+![A screenshot of a cell phone Description automatically
+generated](images/build-pipeline1.png)
+
+![A screenshot of a cell phone Description automatically
+generated](images/build-pipeline2.png)
+
+# Release Pipeline
+
+You can use [this](https://docs.microsoft.com/en-us/azure/devops/pipelines/release/?view=azure-devops) document as a reference on how to create a release pipeline.
+
+1 - Add an [**Azure Deployment tasks**](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-resource-group-deployment?view=azure-devops) for each file generated by extractor, following the order:
```

Comment:
why not use a master template?

Reply:
In my experience with some customers, they prefer use separate templates, because (as they mentioned) it's easy to versioning thing like products in a better way.